app/javascript/admin/controllers/copy_clipboard_controller.js

```diff
@@ -0,0 +1,8 @@
+/* global Runtime */
+import { Controller } from 'stimulus';
+export default class CopyClipboardController extends Controller {
+  handleCopyClick() {
+    const inputValue = document.getElementById('settings-org-secret').value;
+    Runtime.copyToClipboard(inputValue);
```

Comment:
This doesn't seem to be actually copying anything to the clipboard, I've tried to look at some examples and I know the `Runtime.copyToClipboard(inputValue);` is usually followed by `.then()` but I didn't think I needed to handle anything after doing the copying?

Sorry, I've never really used JS before, so maybe I'm missing something simple.

Reply:
Hi @aitchiss, Thanks for the explanation! I'm still a little unsure how to set up the click even listener in the `app/javascript/packs/clipboardCopy.js`  and call it from the button on `_org_admin.html.erb`

I tried this and having the button call this when clicked but that didn't work
```
handleCopyClick() {
  const inputValue = document.querySelector('[data-copy-text-from-input]');
  document.onclick="nextElementSibling.hidden = false";
  Runtime.copyToClipboard(inputValue);
}
```
and just I tried just using the querySelector and then copying but didn't have any luck with that either. 

`document.querySelector('[data-copy-text-from-input]')`

Reply:
I think the best way to do this is:

- Over in `app/views/users/_org_admin.html.erb` add an ID to your button (e.g. `copy-button`)
- You will also need to add an ID to the `<input>` element, e.g. 'secret-code-input'
- In the JS, attach the event listener like:

```js
document.getElementById('copy-button').addEventListener('click', handleCopyClick)
```

And then your handleCopyClick function could look something like:

```js
function handleCopyClick() {
  // Get the value from the input you gave an ID to
  const { value } = document.getElementById('secret-code-input');
  // Copy it to the clipboard, then show the span with the success message
  Runtime.copyToClipboard(value).then(() => {
    document.getElementById('copy-text-announcer').classList.remove('hidden');
  })
}
```

I've not tested this myself but it _should_ work (famous last words!). Let me know if you need any more help 👍